flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/exec/stream/StreamExecRank.java

```diff
@@ -233,60 +240,117 @@ protected Transformation<RowData> translateToPlanInternal(
         long stateRetentionTime =
                 StateMetadata.getStateTtlForOneInputOperator(config, stateMetadataList);
         StateTtlConfig ttlConfig = StateConfigUtil.createTtlConfig(stateRetentionTime);
+        boolean enableAsyncState =
+                config.get(ExecutionConfigOptions.TABLE_EXEC_ASYNC_STATE_ENABLED);
```

Comment:
As a heads up, changes like this should update `consumedOptions` for the JSON serialization.